addons/yatl/chat/chat.gd

```diff
@@ -1,163 +1,171 @@
-tool
-extends Node
-
-
 # Private imports
 
 const __PAL: Resource = preload("../pal/pal.gd")
+const __WebSocketConnection: Resource = __PAL.WebSocketConnection
 
 
-#######
-#
-# TODO: Update this code for new PAL abstraction
-#
-#######
+# Public imports
 
-
-#Custom class preloads
 const ChatMessage = preload("./chat_message.gd")
 
 
-#URL constants
+#Signals
+
+signal connected()
+signal login(success) # success: bool
+signal disconnected()
+signal connection_failed()
+signal channel_joined(channel) # channel: String
+signal channel_left(channel) # channel: String
```

Comment:
If we set this up for being a single channel use, we might be able to get away with simplifying to `connected` + `disconnected`.

Reply:
Addressed in https://github.com/velopteam/YATL/pull/2#discussion_r830534314